keps/sig-windows/20190418-windows-security-context.md

```diff
@@ -112,7 +175,16 @@ type WindowsSecurityOptions struct {
 }
 ```
 
-Field `WindowsOptions *WindowsSecurityOptions` will be added to `SecurityContext` and `PodSecurityContext` structs.
+Field `WindowsOptions *WindowsSecurityContextOptions` will be added to `SecurityContext` and `PodSecurityContext` structs.
+
+#### Validation of fields
+
+The fields within the `WindowsSecurityContextOptions` struct will be validated as follows:
+- GMSACredentialSpecName: This field will be the name of a custom resource. It should follow the rules associated with [naming of resources](https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#names) in Kubernetes. Only the length will be checked to make sure it is below 64 characters.
+- GMSACredentialSpec: The size of the CredentialSpec JSON blob should be limited to avoid abuse. We will limit it to 64K which allows for a lot of room to specify extremely complicated credential specs. Typically this JSON blob is not expected to be more than 1K based on experience so far.
+- RunAsUserName: This field needs to allow for a valid set of usernames allowed for Windows containers. Currently the OCI spec or Windows documentation does not specify any clear restrictions around the length of this parameter or restrictions around usage of special characters when passed to the container runtime and eventually to Windows HCS. So Kubernetes validation won't enforce any character set validation. A maximum length of 64 characters will be allowed to prevent abuse.
```

Comment:
I thought we discussed this as 256 characters?

Reply:
we don't want to make this stricter than what windows would accept

Reply:
Agree. But I am not able to find a clear limit specified anywhere in MSDN or moby/moby today around the above. @PatrickLang any thoughts on ^ ?